Probesys/bileto
# new: Provide a search engine and a syntax

## Related issue(s)



https://github.com/Probesys/bileto/issues/7

## Changes



- Create a Query Tokenizer and Parser to transform a string query into an object
- Create a Ticket Query Builder to transform a Query object in a DQL "where" statement and a list of parameters
- Refactor the TicketSearcher and the TicketRepository to take advantage of the Query system
- Add a search form above the list of tickets

## How to test manually



- Go to the list of tickets
- Change the view and check the list of tickets is correctly filtered
- Also try with different permissions (switch between user / technician roles)
- Filter the tickets with your own queries (e.g. `status:open email`, `-status:open`, `status:new OR no:assignee`, etc.)

## Not done yet

- add the user documentation

## Checklist
